File: MyGithub/app/src/main/java/com/stingerzou/mygithub/di/ActivityBindingModule.java
package com.stingerzou.mygithub.di;

import com.stingerzou.mygithub.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBindingModule {
    @ActivityScoped
    @ContributesAndroidInjector
    abstract MainActivity mainActivity();
}

File: MyGithub/app/src/main/java/com/stingerzou/mygithub/di/ActivityBindingModule.kt
package com.stingerzou.mygithub.di

import com.stingerzou.mygithub.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBindingModule {

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun mainActivity():MainActivity
}